File: Loops/Notes.py
#Evelyn Chennault, Loops Notes Python

#1. What is a loop? 
    #Section of code that repeats multiple times.


#2. What are the two types of loops?
    #for loop
nums = [12,3,66,7,3,3,2]

for num in nums:
    print(num)
x = 0

    #while loop
while x < 10:
    print(x)
    x+= 1


#3. What is iteration?
    #Particular instance/time of the loop
    #Iteration is the amount of times you are looping through something


#4. What are lists? 
    #A group of many things that are related
    #A varible that holds multiple values
    #Array


nums = [1,2,3,4,5,6,7,8]
cousins = ["Riely","Brooklyn", "Tyson", "Troy", "Dylan", "Ruby"]
#print(nums) #prints whole list with brackets and coommas, ugly
print(cousins[3])

cousins.pop()
cousins.insert(0, "Anson")
cousins.insert(2, ["person", "person_two", "person_three"])
print(cousins)


#5. How do you make lists in Python? 
    #Use [] brakcets
    #Add in items with correct data types
    #Have to have a comma between each item

    #YOU CAN HAVE A LIST OF LISTS!!!


#6. How do you make for loops in Python? 
    #
for cousin in cousins:
    print(cousin)

for x in range(1,5):
    print(x)


#7. How do you make while loops in Python?
    #
import random
x = 1 #varible keeps count of my iteration
goose = random.randint(1,20)

while x <= 20:
    if x == goose:
        print("Goose!")
        break #tells the loop to stop
    else:
        print("Burb")
    x+= 1

#continue moves on to the next iteration without finishing

File: Loops/Notes.c
//Evelyn Chennault, Loops Notes C
#include <stdio.h>


int main(void){
    //1. What is a loop? 
        //a section of code repeated multiple times


    //2. What are the two types of loops?
        //while loops
    int start = 0;
    while(start < 5){
        printf("Hello!\n");
        start++;
    }
        //for loops
    int q;
    for(q=0; q<5; q++){
        printf("%d\n", q);
    }


    //3. What is iteration
        //reapeating something with minor changes each time


    //4. What are arrays (lists)?
        //a varible holding multiple values


    //8. How do you make arrays in C?
        //arrays all need to be the same data type
        //1. set the data type
        //2. AFTER naming, put brackets and write the length of the array
        //3. array is surrounded by culry brackets
        //4. commas "," between each item
    int grades[] = {88, 97, 100, 70, 72, 99, 61};
        //print a single int, char, or float
    printf("CSP Grade: %d\n", grades[2]);
        //change item in the array
    grades[2] = 95;
    printf("CSP Grade: %d\n", grades[2]);
        //size of array in bytes
    int size = sizeof(grades);
        //length in items
    int length = sizeof(grades)/sizeof(grades[0]);
    printf("The array is %d items long.\n", length);
        //array of strings
        //forst brackets sets the length of the array
        //second brackets sets length of each string
    char movies[][20] = {"Car", "Treasure Pleanet", "An American Tale", "Marley & Me", "The Avengers"};
    printf("The movie is %s!\n", movies[1]);
    int mlength = sizeof(movies)/sizeof(movies[0]);


    //9. How do you make for loops in C?
        //set the iterator, keeps track of time going through the loop (best practice: use "i" or "x")
    int x;
        //"(starting point; ending point, count by)"
    for(x = 0; x <= 10;x+=2){
        printf("%d\n", x);
    }
    int m;
    for(m = 0;m < mlength; m++){
        printf("%s\n", movies[m]);
    }


    //10. How do you make while loops in C?
    int w = 100; //start

    while(w >= 0){ //stop
        printf("%d\n", w); //count by
        w -= 10;
    }


    return 0;
}